pkg/registry/registry.go

```diff
@@ -54,9 +60,130 @@ func (o RegistryClient) DownloadFileInMemory(params dfutil.HTTPRequestParams) ([
 }
 
 // DownloadStarterProject downloads a starter project referenced in devfile
-// This will first remove the content of the contextDir
+// There are 3 cases to consider here:
+// Case 1: If there is devfile in the starterproject, replace all the contents of contextDir with that of the starterproject; warn about this
+// Case 2: If there is no devfile, and there is no conflict between the contents of contextDir and starterproject, then copy the contents of the starterproject into contextDir.
+// Case 3: If there is no devfile, and there is conflict between the contents of contextDir and starterproject, copy contents of starterproject into a dir named CONFLICT_STARTER_PROJECT; warn about this
 func (o RegistryClient) DownloadStarterProject(starterProject *devfilev1.StarterProject, decryptedToken string, contextDir string, verbose bool) error {
-	return DownloadStarterProject(starterProject, decryptedToken, contextDir, verbose)
+	// Let the project be downloaded in a temp directory
+	starterProjectTmpDir, err := o.fsys.TempDir("", "odostarterproject")
```

Comment:
I think it should be our responsibility to remove this temporary folder when we are done with it. I just noticed I had several non-empty left-over `odostarterproject*` folders on my temporary fielsystem.

Reply:
I figured that since they were temporary files, they would delete themselves on a reboot, but maybe you're right, it would be better to delete them as we go.

Reply:
Yes, upon reboot, but it depends on how frequently you reboot :D